pkg/storage/storage.go

```diff
@@ -61,66 +62,185 @@ func Remove(client *occlient.Client, name string, componentName string, applicat
 func List(client *occlient.Client, componentName string, applicationName string) ([]StorageInfo, error) {
-	labels := storagelabels.GetLabels("", componentName, applicationName, false)
-	selector := util.ConvertLabelsToSelector(labels)
+	componentLabels := componentlabels.GetLabels(componentName, applicationName, false)
+	componentSelector := util.ConvertLabelsToSelector(componentLabels)
 
-	log.Debugf("Looking for PVCs with the selector: %v", selector)
-	pvcs, err := client.GetPVCsFromSelector(selector)
+	dc, err := client.GetOneDeploymentConfigFromSelector(componentSelector)
 	if err != nil {
-		return nil, errors.Wrap(err, "unable to get PVC names")
+		return nil, errors.Wrapf(err, "unable to get Deployment Config associated with component %v", componentName)
+	}
+
+	// store the storages in a map for faster searching with the key instead of list
+	mountedStorageMap := make(map[string]string)
+	volumeMounts := client.GetVolumeMountsFromDC(dc)
+	for _, volumeMount := range volumeMounts {
+		pvcName := client.GetPVCNameFromVolumeMountName(volumeMount.Name, dc)
+		if pvcName == "" {
+			return nil, fmt.Errorf("no PVC associated with Volume Mount %v", volumeMount.Name)
+		}
+		pvc, err := client.GetPVCFromName(pvcName)
+		if err != nil {
+			return nil, errors.Wrapf(err, "unable to get PVC %v", pvcName)
+		}
+
+		storageName := getStorageFromPVC(pvc)
+		mountedStorageMap[storageName] = volumeMount.MountPath
 	}
 
+	pvcs, err := client.GetPVCsFromSelector(storagelabels.StorageLabel)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get Deployment Config associated with component %v", componentName)
+	}
 	var storageList []StorageInfo
 	for _, pvc := range pvcs {
-		storage := getStorageFromPVC(pvc)
-		if storage != "" {
-			size := pvc.Spec.Resources.Requests[corev1.ResourceStorage]
+		pvcComponentName, ok := pvc.Labels[componentlabels.ComponentLabel]
+		pvcAppName, okApp := pvc.Labels[applabels.ApplicationLabel]
+		// first check if storage label does not exists indicating that the storage is not mounted in any component
+		// if the storage label exists, then check if the component is the current active component
+		// also check if the app label exists and is equal to the current application
+		if (!ok || pvcComponentName == componentName) && (okApp && pvcAppName == applicationName) {
+			if pvc.Name == "" {
+				return nil, fmt.Errorf("no PVC associated")
+			}
+			storageName := getStorageFromPVC(&pvc)
+			storageSize := pvc.Spec.Resources.Requests[corev1.ResourceStorage]
+			storagePath := mountedStorageMap[storageName]
 			storageList = append(storageList, StorageInfo{
-				Name: storage,
-				Size: size.String(),
+				Name: storageName,
+				Size: storageSize.String(),
+				Path: storagePath,
 			})
 		}
 	}
 	return storageList, nil
 }
 
+// Exists checks if the given storage exists in the given application
+func Exists(client *occlient.Client, storageName string, componentName string, applicationName string) (bool, error) {
```

Comment:
whi is componentName parameters if it is not used anywhere expect in the error message?